### ISSUE TITLE
Homebrew version of Macvim shows error at start

### ISSUE THREAD

**Issue by cumt559** [Shougo/deoplete.nvim#1211]
**Warning:  I will close the issue without the minimal init.vim and the reproduction instructions.**

# Problems summary

After an upgrade via brew, every time when I open up Evervim I get the following error message:

[vim-hug-neovim-rpc] failed executing: pythonx import [pynvim|neovim]
[vim-hug-neovim-rpc] Vim(pythonx):ModuleNotFoundError: No module named 'neovim'
[deoplete] [vim-hug-neovim-rpc] requires one of `:pythonx import [pynvim|neovim]` command to
 work
[deoplete] VimEnter Autokommandos für "*"..function deoplete#enable[9]..deoplete#initialize[
1]..deoplete#init#_initialize[10]..<SNR>126_init_internal_variables[11]..neovim_rpc#serverad
dr, Zeile 18

I am using Macvim, and it was just upgraded. Before, everything was OK. I have tried to fix the issue following 
https://stackoverflow.com/questions/63024845/exception-not-caught-vim-hug-neovim-rpc-requires-one-of-pythonx-import-pyn

when I input the command ":pythonx import sys; print(sys.path)" I got:

['/opt/homebrew/Cellar/python@3.12/3.12.1_1/Frameworks/Python.framework/Versions/3.12/lib/py
thon312.zip', '/opt/homebrew/Cellar/python@3.12/3.12.1_1/Frameworks/Python.framework/Version
s/3.12/lib/python3.12', '/opt/homebrew/Cellar/python@3.12/3.12.1_1/Frameworks/Python.framewo
rk/Versions/3.12/lib/python3.12/lib-dynload', '/opt/homebrew/Cellar/python@3.12/3.12.1_1/Fra
meworks/Python.framework/Versions/3.12/lib/python3.12/site-packages', '/opt/homebrew/Cellar/
python@3.12/3.12.1_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/
gpg-1.23.2-py3.12-macosx-14-arm64.egg', '_vim_path_']

then I tried the commands 

brew link --overwrite python@3.12 --force
pip3 install pynvim

they simply don't work.

## Environment Information

 * deoplete version (SHA1):

 * OS: macOS 14.2.1 23C71 arm64

 * neovim/Vim `:version` output:

VIM - Vi IMproved 9.1 (2024 Jan 02 kompiliert am Jan 04 2024 03:00:35)
MacOS Version - arm64
Übersetzt von Homebrew
Riesige Version with MacVim GUI.
 Ein- (+) oder ausschließlich (-) der Eigenschaften:
+acl               +find_in_path      +mouse_xterm       +tcl
+arabic            +float             +multi_byte        +termguicolors
+autocmd           +folding           +multi_lang        +terminal
+autochdir         -footer            -mzscheme          +terminfo
-autoservername    +fork()            +netbeans_intg     +termresponse
+balloon_eval      +fullscreen        +num64             +textobjects
+balloon_eval_term +gettext           +odbeditor         +textprop
+browse            -hangul_input      +packages          +timers
++builtin_terms    +iconv             +path_extra        +title
+byte_offset       +insert_expand     +perl              +toolbar
+channel           +ipv6              +persistent_undo   +transparency
+cindent           +job               +popupwin          +user_commands
+clientserver      +jumplist          +postscript        +vartabs
+clipboard         +keymap            +printer           +vertsplit
+cmdline_compl     +lambda            +profile           +vim9script
+cmdline_hist      +langmap           -python            +viminfo
+cmdline_info      +libcall           +python3           +virtualedit
+comments          +linebreak         +quickfix          +visual
+conceal           +lispindent        +reltime           +visualextra
+cryptv            +listcmds          +rightleft         +vreplace
+cscope            +localmap          +ruby              +wildignore
+cursorbind        +lua               +scrollbind        +wildmenu
+cursorshape       +menu              +signs             +windows
+dialog_con_gui    +mksession         +smartindent       +writebackup
+diff              +modify_fname      +sodium            -X11
+digraphs          +mouse             +sound             -xattr
+dnd               +mouseshape        +spell             -xfontset
-ebcdic            +mouse_dec         +startuptime       +xim
+emacs_tags        -mouse_gpm         +statusline        -xpm
+eval              -mouse_jsbterm     -sun_workshop      -xsmp
+ex_extra          +mouse_netterm     +syntax            -xterm_clipboard
+extra_search      +mouse_sgr         +tag_binary        -xterm_save
-farsi             -mouse_sysmouse    -tag_old_static
+file_in_path      +mouse_urxvt       -tag_any_white
          System-vimrc-Datei: "$VIM/vimrc"
        Benutzer-vimrc-Datei: "$HOME/.vimrc"
 zweite Benutzer-vimrc-Datei: "~/.vim/vimrc"
         Benutzer-exrc-Datei: "$HOME/.exrc"
         System-gvimrc-Datei: "$VIM/gvimrc"
       Benutzer-gvimrc-Datei: "$HOME/.gvimrc"
zweite Benutzer-gvimrc-Datei: "~/.vim/gvimrc"
      defaults Datei: "$VIMRUNTIME/defaults.vim"
           System-Menü-Datei: "$VIMRUNTIME/menu.vim"
     Voreinstellung für $VIM: "/Applications/MacVim.app/Contents/Resources/vim
"
Übersetzt: clang -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_MACVIM -Wall -Wno-unknown-pragmas -pipe -DMACOS_X -DMACOS_X_DARWIN -g -O2 -arch arm64 -D_REENTRANT -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linken: clang -arch arm64 -L/opt/homebrew/lib -o Vim -lm -lncurses /opt/homebrew/lib/libsodium.a -liconv /opt/homebrew/lib/libintl.a -framework AppKit -L/opt/homebrew/opt/lua/lib -llua5.4 -L/System/Library/Perl/5.34/darwin-thread-multi-2level/CORE -lperl -L/opt/homebrew/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/lib/python3.12/config-3.12-darwin -lpython3.12 -framework CoreFoundation -F/System/Library/Frameworks -framework Tcl -framework CoreFoundation -lruby.3.3 -L/opt/homebrew/Cellar/ruby/3.3.0/lib

 * `:checkhealth` result(neovim only):

## Provide a minimal init.vim/vimrc with less than 50 lines (Required!)

```vim
" Your minimal init.vim/vimrc
set runtimepath+=~/path/to/deoplete.nvim/
let g:deoplete#enable_at_startup = 1

if !NVIM()
    Plug 'roxma/nvim-yarp'
    Plug 'roxma/vim-hug-neovim-rpc'
endif
if (has('python3') == 0)
    echoerr "Deoplete needs NeoVim with +Python3 support!"
endif
Plug 'Shougo/deoplete.nvim', { 'do': ':UpdateRemotePlugins' }
Plug 'Shougo/neco-syntax'
"Plug 'autozimu/LanguageClient-neovim'
Plug 'SirVer/ultisnips'
Plug 'honza/vim-snippets'

if isdirectory(expand(EverVimBundleDir("deoplete.nvim")))
    let g:deoplete#enable_at_startup = 1
    call deoplete#custom#option({
    \ 'auto_complete_delay': 100,
    \ 'smart_case': v:true,
    \ })
    call deoplete#custom#var('omni', 'input_patterns', {
      \ 'tex': g:vimtex#re#deoplete
      \})
    inoremap <silent><expr> <S-TAB> pumvisible() ? "\<C-p>" : "\<S-TAB>"
    inoremap <silent><expr> <Tab> pumvisible() ? "\<C-n>" : "\<Tab>"
endif


" For Vim only
"set runtimepath+=~/path/to/nvim-yarp/
"set runtimepath+=~/path/to/vim-hug-neovim-rpc/
```

## How to reproduce the problem from neovim/Vim startup (Required!)

which python3                                
/usr/bin/python3

python3 --version                              
Python 3.9.6

echo $PATH                                  
/Users/cumt559/Library/Python/3.9/bin:/opt/local/bin:/opt/local/sbin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/TeX/texbin

pip3 install --user --upgrade pynvim         

Requirement already satisfied: pynvim in /Users/cumt559/Library/Python/3.9/lib/python/site-packages (0.5.0)
Requirement already satisfied: msgpack>=0.5.0 in /Users/cumt559/Library/Python/3.9/lib/python/site-packages (from pynvim) (1.0.7)
Requirement already satisfied: greenlet>=3.0 in /Users/cumt559/Library/Python/3.9/lib/python/site-packages (from pynvim) (3.0.3)

## Generate a logfile if appropriate

 1. export NVIM_PYTHON_LOG_FILE=/tmp/log
 2. export NVIM_PYTHON_LOG_LEVEL=DEBUG
 3. nvim -u minimal.vimrc
 4. some works
 5. cat /tmp/log_{PID}

## Screenshot (if possible)
![image](https://github.com/Shougo/deoplete.nvim/assets/61304428/d122bdd2-9a81-43f1-a756-9179655185ef)

**Comment by Shougo** [comment]
It is not the issue of deoplete.

`pynvim` is not installed in `your neovim's Python`.

```
then I tried the commands

brew link --overwrite python@3.12 --force
pip3 install pynvim

they simply don't work.
```

You need to check what `pip3` is executed.
I think `python3 import pynvim` does not work in your neovim.


```
which python3
/usr/bin/python3

python3 --version
Python 3.9.6

echo $PATH
/Users/cumt559/Library/Python/3.9/bin:/opt/local/bin:/opt/local/sbin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/TeX/texbin

pip3 install --user --upgrade pynvim

Requirement already satisfied: pynvim in /Users/cumt559/Library/Python/3.9/lib/python/site-packages (0.5.0)
Requirement already satisfied: msgpack>=0.5.0 in /Users/cumt559/Library/Python/3.9/lib/python/site-packages (from pynvim) (1.0.7)
Requirement already satisfied: greenlet>=3.0 in /Users/cumt559/Library/Python/3.9/lib/python/site-packages (from pynvim) (3.0.3)
```

Your Python 3 commands(and pip3) is Python 3.9.  But your neovim is not.

Because of

```
Übersetzt: clang -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_MACVIM -Wall -Wno-unknown-pragmas -pipe -DMACOS_X -DMACOS_X_DARWIN -g -O2 -arch arm64 -D_REENTRANT -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linken: clang -arch arm64 -L/opt/homebrew/lib -o Vim -lm -lncurses /opt/homebrew/lib/libsodium.a -liconv /opt/homebrew/lib/libintl.a -framework AppKit -L/opt/homebrew/opt/lua/lib -llua5.4 -L/System/Library/Perl/5.34/darwin-thread-multi-2level/CORE -lperl -L/opt/homebrew/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/lib/python3.12/config-3.12-darwin -lpython3.12 -framework CoreFoundation -F/System/Library/Frameworks -framework Tcl -framework CoreFoundation -lruby.3.3 -L/opt/homebrew/Cellar/ruby/3.3.0/lib
````

It is Python 3.12!
You need to install `pynvim` package in Python 3.12.

**Comment by cumt559** [comment]
>
> Your Python 3 commands(and pip3) is Python 3.9. But your neovim is not.
>
> It is Python 3.12!
> You need to install pynvim package in Python 3.12.
>
Understood. I have tried the following:

***@***.***/3.12.1_1/bin/pip3.12 install pynvim


error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try brew install
    xyz, where xyz is the package you are trying to
    install.

    If you wish to install a non-brew-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip.

    If you wish to install a non-brew packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.

note: If you believe this is a mistake, please contact your Python
installation or OS distribution provider. You can override this, at the
risk of breaking your Python installation or OS, by passing
--break-system-packages.
hint: See PEP 668 for the detailed specification.

[notice] A new release of pip is available: 23.3.1 -> 23.3.2
[notice] To update, run: python3.12 -m pip install --upgrade pip



Do you know how to fix this?

Many thanks.

**Comment by cumt559** [comment]
I bravely used the  --break-system-packages flag, now it seems to be
working fine...


× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
python3-xyz, where xyz is the package you are trying to
install.

If you wish to install a non-Debian-packaged Python package,
create a virtual environment using python3 -m venv path/to/venv.
Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
sure you have python3-full installed.

If you wish to install a non-Debian packaged Python application,
it may be easiest to use pipx install xyz, which will manage a
virtual environment for you. Make sure you have pipx installed.


Am Do., 25. Jan. 2024 um 09:42 Uhr schrieb Shougo ***@***.***
